pkg/csi/plugins/nodeserver.go

```diff
@@ -167,6 +167,16 @@ func (ns *nodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 func (ns *nodeServer) NodeUnpublishVolume(ctx context.Context, req *csi.NodeUnpublishVolumeRequest) (*csi.NodeUnpublishVolumeResponse, error) {
 	targetPath := req.GetTargetPath()
 
+	// check path existence
+	_, err := os.Stat(targetPath)
+	if err != nil {
+		return nil, errors.Wrapf(err, "NodeUnpublishVolume: stat targetPath %s error %v", targetPath, err)
+	}
+
+	// The lock is to avoid race condition
+	ns.mutex.Lock()
+	defer ns.mutex.Unlock()
```

Comment:
Please consider two things:

1. The lock scope is too wide, we'd better to restrict it to mount path level.
2. Is the lock only for NodeUnpublishVolume?

Reply:
Have modified: use mount path level lock and also lock for NodePublishVolume